fix(components): [date-picker-panel] remove incorrect is-disabled class from range panel header

shanecranor opened the pull request:
closed #23102

This PR fixes some bad logic for determining when to apply the is-disabled class. The behavior prior to this PR included `!enableYearArrow`, however the purpose of `!enableYearArrow` seems to be to disable the "Previous/Next Year" button and not to disable the entire header.

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

shanecranor committed:
fix: remove incorrect is-disabled class from range panel header

closed #23102

shanecranor committed:
Merge branch 'dev' into fix/23102-date-picker-header-broken

shanecranor committed:
Merge branch 'dev' into fix/23102-date-picker-header-broken

shanecranor committed:
Merge branch 'dev' into fix/23102-date-picker-header-broken

Dsaquel commented on packages/components/date-picker-panel/src/date-picker-com/panel-date-range.vue:
```diff
@@ -244,10 +244,7 @@
-              drpNs.e('header'),
-              ppNs.is('disabled', !enableYearArrow || dateRangeDisabled),
-            ]"
+            :class="[drpNs.e('header'), ppNs.is('disabled', dateRangeDisabled)]"
```
The class can be deleted completely it has in fact no effect at all.
```suggestion
            :class="drpNs.e('header')"
```

shanecranor replied:
Agreed, lets delete it then. I wasn't sure if there was a reason it was there.

shanecranor committed:
Update packages/components/date-picker-panel/src/date-picker-com/panel-date-range.vue

Co-authored-by: Noblet Ouways <91417411+Dsaquel@users.noreply.github.com>

shanecranor committed:
noissue: Remove disabled class test for date picker header

Removed a test case that checks if the date picker panel header has the 'is-disabled' class when enabled.

Dsaquel committed:
style: fmt

Dsaquel reviewed (approved):
Thank you
